docs/DesignDocs/Attestation_API_Proposal.md

```diff
@@ -0,0 +1,599 @@
+plugin libraries, composed of:
+- One enclave-side SGX evidence verification plugin library.
+  - It implements the enclave-side OE SDK public API function
+  `oe_verifier_initialize()` for enclave-side verifier plugins enumeration
+  and registration, and `oe_verifier_shutdown()`.
+- One host-side SGX evidence verification plugin library.
+  - It implements the host-side OE SDK public API function
+  `oe_verifier_initialize()` for host-side verifier plugins enumeration
+  and registration, and `oe_verifier_shutdown()`.
+  - It also implements OCALLs for the enclave-side library to access
+  host-side services for SGX quote verification.
+
+Note: host-side SGX plugin library implementation of the public API functions
+`oe_verifier_initialize()` and `oe_verifier_shutdown()` indicates that
+this library can't be linked to an verifier application along with
+other plugin libraries that implement the same public API functions.
+This limitation will be relaxed in a future release when a more sophisticated
+mechanism is adopted for plugin library initialization.
+More general discussion on evidence formats, plugins, plugin libraries
+and their initialization can be found in the "Notes on SGX Plugins Design"
+section of the [[Design Notes]](https://github.com/shnwc/openenclave/blob/master/docs/DesignDocs/NotesOnAttestationAPI.md#notes-on-sgx-plugins-design)
+document.
+
+Verification of an SGX report for local attestation can only be performed by
+the enclave to which the SGX report is targeted. So no host-side plugin
+for SGX local attestation evidence can be supported.
+
+Verification of SGX EPID evidence (or quote) is not supported by the
+Intel SGX SDK. In existing SGX solutions, EPID quotes are sent from an
+attester enclave to a relying party, which forwards the EPID quotes to
+an Intel Attestation Service (IAS) for verification. There is no plan
+in the OE SDK V0.10 release to provide EPID evidence verifier plugin.
+
+The OE SDK has legacy API functions `oe_get_report()` and `oe_verify_report()`
+for generation and verification of SGX local and ECDSA reports.
+If support of these legacy API functions is required, then the plugins for
+SGX ECDSA and local evidence formats needs to implement the entry points
+`(*get_report)(...)` and `(*verify_report)(...)`, and the OE SDK framework needs
+to implement the legacy functions `oe_get_report()` and `oe_verify_report()` on
+these entry points respectively.
+
+The Legacy functions `oe_get_report()` and `oe_verify_report()` should be
+implemented by the OE SDK framework to first call API functions
+`oe_attester_initializat()` and `oe_verifier_initialize()` respectively,
+since legacy applications don't call these API functions explicitly.
+ As `oe_attester_initializat()` and `oe_verifier_initialize()` are
+ idempotent, there is no harm calling them multiple times. For SGX plugins,
+ `oe_attester_shutdown()` and `oe_verifier_shutdown()` are NOP, so there
+ is no need to call them upon application close.
```

Comment:
```suggestion
The Legacy functions `oe_get_report()` and `oe_verify_report()` should be
implemented by the OE SDK framework to first call API functions
`oe_attester_initializat()` and `oe_verifier_initialize()` respectively,
since legacy applications don't call these API functions explicitly.
 As `oe_attester_initializat()` and `oe_verifier_initialize()` are
 idempotent, there is no harm calling them multiple times. For SGX plugins,
 `oe_attester_shutdown()` and `oe_verifier_shutdown()` are NOP, so there
 is no need to call them upon application close.
```
```suggestion
The Legacy functions `oe_get_report()` and `oe_verify_report()` should be
implemented by the OE SDK framework to first call API functions
`oe_attester_initialize()` and `oe_verifier_initialize()` respectively,
since legacy applications don't call these API functions explicitly.
 As `oe_attester_initialize()` and `oe_verifier_initialize()` are
 idempotent, there is no harm calling them multiple times. For SGX plugins,
 `oe_attester_shutdown()` and `oe_verifier_shutdown()` are NOP, so there
 is no need to call them upon application close.
```

Reply:
@jhand2 thank you for catching these strange typos! Just committed the fix.

Reply:
@jhand2 my last commit had the suggested edits, and was squashed and signed off. Let me know If I didn't do the squash in the correct way.